### PR TITLE
[StructuralMechanicsApplication] Rigid movement process

### PR DESCRIPTION
**Pull request by loumalouomega**
This adds the possibility to consider a rigid body movement in a certain region of the problem
    
This process constructs the model parts containing the constrains that enforce the rigid body movement

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.h
Comment (adityaghantasala) [inline applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.h:257]: Why commenting out this part ?

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.h:257]: I don't remember...

File: applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py
Comment (adityaghantasala) [inline applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: Why is this new_model_part_name required ? can't there be just a computing_mp and the model_part_name (which is rigid) and we can directly add the constraints to the computing_modelpart ?

File: applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py
Comment (adityaghantasala) [inline applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: why is rigid_model_part different from the model_part from line 62 ?

File: applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp
Comment (adityaghantasala) [inline applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.cpp]: following how you defined above ... here `thread_id`  instead of just  `id` is more apt and easy to follow.

**Comment by loumalouomega** [comment]
Anyone?

File: applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: All the process that affect the system of dof require the computing model part, because you pass the model and the model does not know the computing model part (at least now)

File: applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: rigid_model_part->new_model_part_name The model part created
computing_model_part->computing_model_part_name The computing model aprt, that the model not know now
model_part->model_part_name The destination model part, this is the only one you need to set to work. The computing model part is usually called computing_domain, so no problem. The new model part is just a name for the new model part, not so important

File: applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py
Comment (adityaghantasala) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: Yes I see why the computing model part name is required ! 
But, my question (together with comment below) is : why to create an additional modelpart named **Rigid_Movement_ModelPart** instead we can already work on **model_part->model_part_name** and computing modelpart. Thus we do not require the transfer_process

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: Because maybe I use that model part for other pourposes, I like to have everything in order

Comment (adityaghantasala) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: Ok ... fine. My only concern is that this is extra work with fast transfer between modelparts and is complicated to follow with different modelparts  ... 
When this condition is added to the GiD interface we will have an unique and exclusive submodelpart for this condition so no need and this is extra work. Also when passing a submodelpart to this process, it is evident that process will add constraints (from the comments in the code you made).  
Also in the cases where this submodelpart is used somewhere else, this sub model part is not being modified (in terms of nodes and elements) so  I do not see why it is a problem.

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: What do you suggest?, to use the origin model part?

Comment (adityaghantasala) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: Yes the modelpart with "model_part_name" as name and as you correctly pointed out (which I missed) the computing modelpart !

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: OK

Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py]: Done, I consider the origin model part as destination now by default

**Comment by adityaghantasala** [comment]
I checked the other process in the core .. the overload of << and >> operators exists. We can uncomment this part of  to be consistent !! Then everything is a go from my side ! 
Impose_rigid_movement_process.h 
Other wise .. all good for me. 

@loumalouomega  Thank you for using the `model_part_name` modelpart. 

Regarding moving this process into core, may be a with a different name as `rigid_movement`  somehow points to structure related behavior.

File: applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.h
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_processes/impose_rigid_movement_process.h:257]: Done

**Comment by loumalouomega** [comment]
Done

File: applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py
Comment (adityaghantasala) [inline applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py:36]: This "new_model_part_name" can be gone. It just is not clear to follow why is this required and it is easy to get confused with different modelparts here ... 
Can you leave a comment in the code before merging ?

**Review by adityaghantasala — DISMISSED** [review]

File: applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/python_scripts/impose_rigid_movement_process.py:36]: It is empty by default, and some people->me, likes to add the constraints to a differerent model part

**Comment by loumalouomega** [comment]
Can anyone approve?

**Comment by adityaghantasala** [comment]
I approved no ? I can see this is approved !

**Comment by philbucher** [comment]
it is approved but there are conflicts that need to be resolved

**Comment by adityaghantasala** [comment]
perfect ... I will approve again once the checks are finished !

**Comment by loumalouomega** [comment]
It compiled

**Review by adityaghantasala — APPROVED** [review]
all good ... approving !

**Comment by loumalouomega** [comment]
Thanks

**Pull request merged**
